### ISSUE TITLE
Support on TPU for tf.contrib.framework.sort

### ISSUE THREAD

**Issue by 1vn** [tensorflow/tensorflow#19517]
We tried to train with TPU on our model which needs to sort a tensor along an axis. Using  tf.contrib.framework.sort results in the following error:
```
NotFoundError (see above for traceback): No registered 'TopKV2' OpKernel for XLA_TPU_JIT devices compatible with node TPUReplicate/loop/resnet/unit_2_4/sub1/conv1/sort/TopKV2 = TopKV2[T=DT_FLOAT, sorted=true, _device="/device:TPU_REPLICATED_CORE"](TPUReplicate/loop/resnet/unit_2_4/sub1/conv1/sort/transpose, TPUReplicate/loop/resnet/unit_2_4/sub1/conv1/sort/strided_slice)
```

Are there any alternatives? 

Have I written custom code: N/A
OS Platform and Distribution: ubuntu
TensorFlow installed from: official
TensorFlow version: N/A
Bazel version: N/A
CUDA/cuDNN version: N/A
GPU model and memory: TPU on gcloud
Exact command to reproduce: any training script using tf.contrib.framework.sort

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by 1vn** [comment]
yes

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 63 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 94 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 109 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 124 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 139 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 154 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 169 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by mohantym** [comment]
Hi! @1vn ,
It seems that  you are using old version of Tensorflow which is officially considered as end of life, We recommend that you upgrade to 2.4 or later version and let us know if the issue still persists in newer versions .Please open a new issue  with a sample code in case you face any errors, we will get you the right help .Thanks!
